ui/src/components/chat/compose.rs

```diff
@@ -343,25 +343,34 @@ fn get_chatbar(cx: Scope<ComposeProps>) -> Element {
             )),
             with_replying_to: data.map(|data| {
                 let active_chat = data.active_chat.clone();
+                let participants = build_participants(&data.active_chat.participants.clone());
+
                 cx.render(rsx!(
-                    active_chat.clone().replying_to.map(|msg| rsx!(
+                    active_chat.clone().replying_to.map(|msg|
+                    {
+                    let our_did = state.read().account.identity.did_key();
```

Comment:
what if instead of checking `our_did` and `their_did`, we took the vector of participants and called `vec.retain(|p| p.did_key() == msg.sender())` and then called `vec.iter().next()` to get the first item in the filtered list. this eliminates the need to know who is where in the vector (which could change, especially for group messages)